src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -2757,6 +2758,53 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Sve_CreateFalseMaskByte:
+        case NI_Sve_CreateFalseMaskDouble:
+        case NI_Sve_CreateFalseMaskInt16:
+        case NI_Sve_CreateFalseMaskInt32:
+        case NI_Sve_CreateFalseMaskInt64:
+        case NI_Sve_CreateFalseMaskSByte:
+        case NI_Sve_CreateFalseMaskSingle:
+        case NI_Sve_CreateFalseMaskUInt16:
+        case NI_Sve_CreateFalseMaskUInt32:
+        case NI_Sve_CreateFalseMaskUInt64:
+        {
+            // Import as a constant vector 0
+            retNode = gtNewZeroConNode(retType);
+            break;
+        }
+
+        case NI_Sve_CreateTrueMaskByte:
+        case NI_Sve_CreateTrueMaskDouble:
+        case NI_Sve_CreateTrueMaskInt16:
+        case NI_Sve_CreateTrueMaskInt32:
+        case NI_Sve_CreateTrueMaskInt64:
+        case NI_Sve_CreateTrueMaskSByte:
+        case NI_Sve_CreateTrueMaskSingle:
+        case NI_Sve_CreateTrueMaskUInt16:
+        case NI_Sve_CreateTrueMaskUInt32:
+        case NI_Sve_CreateTrueMaskUInt64:
+        {
+            assert(sig->numArgs == 1);
+
+            op1 = impPopStack().val;
+
+            if (op1->IsIntegralConst(31))
+            {
+                // This is considered to be an all true mask. Import as a constant vector all bits set.
+                // TODO: Depending on the vector length, we may be able to consider other patterns
+                // as all true mask, however these will not be commonly used.
+                retNode = gtNewAllBitsSetConNode(retType);
+            }
+            else
+            {
+                // Create a node of TYP_MASK, making sure to update pRetType
+                retNode   = gtNewSimdHWIntrinsicNode(TYP_MASK, op1, intrinsic, simdBaseJitType, simdSize);
+                *pRetType = TYP_MASK;
```

Comment:
We should be able to import a vector constant for any case where `op1` is constant, correct?

Does the handling need to be centralized and also called from rationalization (and potentially morph) so we can catch any inputs that become constant later?

Reply:
> We should be able to import a vector constant for any case where `op1` is constant, correct?
> 
> Does the handling need to be centralized and also called from rationalization (and potentially morph) so we can catch any inputs that become constant later?

Only for constant 31. All other values aren't a fully all bits set vector.

I didn't think about values that aren't currently constant. If we wanted to handle those, what would be imported here? A TrueMask which is then turned into constant vector in rationalization/morph?

Reply:
As per the above, you'd introduce an appropriate vector constant representing the mask where appropriate.

For the JIT and some AOT configurations this is trivial as we know the SVE vector size. For other AOT configurations, we don't know the size and so would need to keep it as a HWIntrinsic node for that scenario.